File: leet_DP_128_lcs___hashing.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums)
    {
        vector<int>     dp(len, 1);
        int             len, i, j, tmp, dif;

        len = (int) nums.size();
        if (len < 2)
            return (len);
        sort(nums.begin(), nums.end());
        i = 0;
        while (++i < len)
        {
            dif = nums[i] - nums[i - 1];
            if (dif < 2)
                dp[i] = dp[i - 1] + dif;
        }
        return (*std::max_element(dp.begin(), dp.end()));
    }
};

File: leet_DP_128_lcs___hashing.py
class Solution:
    def longestConsecutive(self, n: List[int]) -> int:
        L = len(n)
        if L < 2:
            return L
        E = [1] * L
        n.sort()
        for i in range(1, L):
            t = n[i] - n[i - 1]
            if t < 2:
                E[i] = E[i - 1] + t
        return max(E)
